app/routes/customer_routes.py

```diff
@@ -0,0 +1,153 @@
+            "phone": customer.phone,
+        })
+    return jsonify(customer_response)
+
+
+@customers_bp.route("/<customer_id>", methods=["GET"])
+def get_one_customer(customer_id):
+
+    if customer_id.isnumeric() != True:
+        return jsonify({"error": "Invalid Data"}), 400
+
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+        # refactor with helper function
+    response_body = {
+        "id": customer.customer_id,
+        "name": customer.name,
+        "registered_at": customer.registered_at,
+        "postal_code": customer.postal_code,
+        "phone": customer.phone}
+    return jsonify(response_body), 200
+
+
+@customers_bp.route("/<customer_id>", methods=["PUT"])
+def update_one_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    request_body = request.get_json()
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    if "name" not in request_body or type(customer.name) != str:
+        return make_response("", 400)
+    if "postal_code" not in request_body or type(customer.postal_code) != str:
+        return make_response("", 400)
+    if "phone" not in request_body or type(customer.phone) != str:
+        return make_response("", 400)
+
+    customer.name = request_body["name"]
+    customer.postal_code = request_body["postal_code"]
+    customer.phone = request_body["phone"]
+
+    db.session.add(customer)
+    db.session.commit()
+
+    response_body = {
+        "id": customer.customer_id,
+        "name": customer.name,
+        "registered_at": customer.registered_at,
+        "postal_code": customer.postal_code,
+        "phone": customer.phone,
+    }
+
+    return jsonify(response_body), 200
+
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_one_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    db.session.delete(customer)
+    db.session.commit()
+
+    return {"id": customer.customer_id}
+
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def customer_rentals(customer_id):
+    if customer_id.isnumeric() != True:
+        return jsonify({"error": "Invalid Data"}), 400
+
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    rentals = Rental.query.filter_by(
+        customer_id=customer.customer_id, videos_checked_in=False)
+
+    response_body = list()
+
+    for rental in rentals:
+        video = Video.query.get(rental.video_id)
+
+        response_body.append(
+            {
+                "release_date": video.release_date,
+                "title": video.title,
+                "due_date": rental.due_date})
+
+    return jsonify(response_body), 200
```

Comment:
Queries have resource cost (mostly time due to how small the dataset it) but as we scale our applications we want to consider the speed/memory it takes to perform tasks. 

In this case, we can reduce the amount of queries (aka the `get()` method) from 3 to 1 by utilizing the `videos` attribute in `Customer`. `customer.videos` derives from one-to-many relationship Customer has with rentals. Using `customer.videos` provides all the videos that the customer has rented in the form of a list of video instances. 

Using a helper dictionary in the `Video` to turn video data in dictionary form, we can build the response body  for this function like so:

```
response_body = [video.to_dict() for video in customer.videos]
```